### DISCUSSION TITLE
Which files are parsed as tabs on the repo's front page?

### DISCUSSION THREAD

**Discussion by erosman** [community/community#82712] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

GitHub creates tabs for some files on the repo front page.

So far, I have seen these files are parsed into tabs:
- CODE_OF_CONDUCT.md
- LICENSE
- README.md
- SECURITY.md

![image](https://github.com/community/community/assets/6458876/9fb70e5a-562a-44f2-bd73-3cbf718fa556)

What other files are parsed that way?
It would be great if other files such as `PRIVACY.md` could be parsed.

**Comment by faridnec** [comment, marked as answer]
Files currently parsed into tabs are
1. `README.md` (always parsed as the first tab)
2. `CODE_OF_CONDUCT.md`
3. `LICENSE.md` 
4. `SECURITY.md`

but yeah I agree `PRIVACY.md` should be added

**Reply by ADTC** [reply]
Please also add `CONTRIBUTING.md` as a tab. (This is a feature request. It doesn't actually work right now.)

**EDIT (May 30, 2025):** Please make it user-configurable under `.github` folder, so we can just control which files are displayed, in what order, and with what tab names.

A simple CSV file like this would suffice: `.github/tabs.csv`

```csv
README.md
CODE_OF_CONDUCT.md
LICENSE.txt, MIT License
SECURITY.md, "Security, Safety"
CONTRIBUTING.md, Contribution Guidelines
someFolder/UPGRADE.md, How to Upgrade
```

The order of tabs is the same order as the lines, and the optional title comes after the comma. Ignore lines where the first column value doesn't resolve to an existing MD or TXT file.

There could also be a third optional column to indicate a custom anchor link name: `README.md, , #instructions`

(⚠️ Note: This is a feature request. It doesn't actually work right now.)

**Reply by yCodeTech** [reply]
Please add `UPGRADE.md` as a tab too. This would be a better UX than searching the files to help users upgrade certain things from version to version.

**Comment by erosman** [comment]
`FAQ.md` is another worthy candidate.

**Reply by kebaldwi** [reply]
like this one

**Comment by erosman** [comment]
I had a quick search and `CONTRIBUTING.md` & `CHANGELOG.md` do not appear in the tabs.

- https://github.com/rails/rails
- https://github.com/standard/standard

**Reply by faridnec** [reply]
thanks for the correction

**Comment by creizlein** [comment]
Btw, CONTRIBUTING.md seems not to be parsed as a new tab

**Reply by faridnec** [reply]
thanks for the correction

**Comment by erosman** [comment]
Will there be a change in the parsing?
Is it being considered?

**Comment by Ryder17z** [comment]
Sometimes, it would also be nice to have a "BUILD.md" option too.

**Reply by yokwejuste** [reply]
By `BUILD.md` do you mean a file to keep track of the recent builds and or version releases of the software?

Then it will great to look after: [this blog post from FreeCodeCamp](https://www.freecodecamp.org/news/a-beginners-guide-to-git-what-is-a-changelog-and-how-to-generate-it/)

**Reply by Ryder17z** [reply]
Build instructions or that. Depends on the repo in the few cases such a file already exists.

**Comment by VigneshVSV** [comment]
It would be great to give the repository owner/manager the option to choose which markdown files should be shown. Say, I like to have Examples.MD to be shown alongside readme.

**Reply by yokwejuste** [reply]
I think the option is given to you somehow already.

Curiosity brought me to this issue :)

With those files:
- `README.md`
- `CODE_OF_CONDUCT.md`
- `LICENSE.md`
- `SECURITY.md`

What I will give as advice to @VigneshVSV is to only create the necessary ones or the ones you will need or want it to appear on the repo page.

**Reply by VigneshVSV** [reply]
Hi, thanks for the reply.

I am not sure I understood your suggestion because how can I not have these files? I need license, code of conduct, security etc. for my repository to be usable by others. 

What i meant was that if one has extra files that need to turn up on the top, GitHub could just simply add a feature for custom named markdown files to be shown on top.

In the particular case of BUILD.md which the OP wants, he can simply name that file and it turns up on the top. I was requesting for a generic solution.

**Reply by yokwejuste** [reply]
I get this now. Including the license brings a bigger challenge now.

maybe some displays by the side to select which one can be used will be fine.

**Reply by kebaldwi** [reply]
Release as in RELEASE.md should be added

**Comment by thomasmoon** [comment]
+1 For having user configurable tab pages, I'd like to use these for different sections of a style guide for easy browsing right in GitHub.
